internal/databases/postgres/extract_provider_with_spec.go

```
@@ -0,0 +1,90 @@
+package postgres
+
+import (
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/wal-g/wal-g/internal"
+)
+
+const (
+	defaultTbspPrefix = "/" + DefaultTablespace + "/"
+	customTbspPrefix  = "/" + TablespaceFolder + "/"
+	systemIDLimit     = 16384
+)
+
+type ExtractProviderDBSpec struct {
+	ExtractProviderImpl
+	onlyDatabases []string
+}
+
+func NewExtractProviderDBSpec(onlyDatabases []string) *ExtractProviderDBSpec {
+	return &ExtractProviderDBSpec{onlyDatabases: onlyDatabases}
+}
+
+func (p ExtractProviderDBSpec) Get(
+	backup Backup,
+	filesToUnwrap map[string]bool,
+	skipRedundantTars bool,
+	dbDataDir string,
+	createNewIncrementalFiles bool,
+) (IncrementalTarInterpreter, []internal.ReaderMaker, string, error) {
+	_, filesMeta, _ := backup.GetSentinelAndFilesMetadata()
```

do not ignore this error